### ISSUE TITLE
AutoHighlight rules do not auto-activate based on URL conditions

### ISSUE THREAD

**Issue by Rekur** [fastaddons/AutoHighlight#8]
It appears that AutoHighlight doesn't auto-activate certain rules once Firefox is restarted.

Rules that are based on the condition "Doman is" seem to auto-activate just fine. Rules that only have a single condition "URL is" or "URL contains" do not seem to get auto-activated (at least not always). (Once manually activated via the context menu entry "Auto Highlight on this page", the selected rule becomes active and works fine.)

AutoHighlight 2.7, Firefox 88.0.1 x64, Win7/SP1/x64

P.S. Thank you for your excellent add-on!

**Comment by fastaddons** [comment]
Does the rule gets applied when you refresh the page?
Or is the problem only for the initial page load after browser was restarted? Some pages may load faster than the addon so those may be missed. Is this the case?

**Comment by Rekur** [comment]
> Does the rule gets applied when you refresh the page?

Yes, as a matter of fact it does (details below).

I am mainly seeing non-auto-applied rules on [Inoreader](https://www.inoreader.com) page with RSS feeds, and that page loads very slowly the first time. What I see is this: re-launch Firefox; Inoreader page opens; rule with "URL is" match type is not auto-applied (verified in AutoHighlight panel) even though the URL matches; refresh page; same rule is now applied.

However, only the above rule for the currently active URL is applied. If I navigate in Inoreader to another newsfeed, the URL gets updated reflecting that, but another rule set up with "URL is" condition does not get auto-activated. All rules set up with "Domain is" are auto-activated, of course.

OTOH, I have just tested with example.com (which loads very quickly) and "URL contains", and I have not observed any problems there.

Let me know if there are any further tests I can perform on my end. Thanks!

**Comment by Juraj-Masiar** [comment]
Is this issue still present?
I've made a lot of changes in the meantime so maybe it works now?
I know some addons may not be "ready" when Firefox starts.

**Comment by Rekur** [comment]
> Is this issue still present?

Yes, I'm afraid it still is. I just checked on v4.3. For the Inoreader URL that was active when I launched Firefox, rules with "URL is" that matches that initial URL got auto-activated properly as expected. However, when Inoreader "advanced" to another folder and updated the URL accordingly, other rules with matching "URL is" did not get activated as they should have.

Thank you for staying on top of things!
